planning/behavior_velocity_planner/src/scene_module/detection_area/scene.cpp

```diff
@@ -203,6 +203,68 @@ bool DetectionAreaModule::modifyPathVelocity(PathWithLaneId * path, StopReason *
   return true;
 }
 
+// calc smallest enclosing circle with average O(N) algorithm
+std::pair<lanelet::BasicPoint2d, double> calcSmallestEnclosingCircle(
```

Comment:
@maxime-clem 
can you check this algorithm too?

Reply:
@veqcc can you add a reference to the algorithm you implemented ?
I only searched a little but it seems the state of art algorithm is [Megiddo's prune-and-search algorithm](https://www.cs.mcgill.ca/~cs507/projects/1998/jacob/solutions.html) with O(n) complexity.

Reply:
I found the reference (https://people.inf.ethz.ch/emo/PublFiles/SmallEnclDisk_LNCS555_91.pdf).
This algorithm is called "Welzl's algorithm".
```
Welzl, Emo. "Smallest enclosing disks (balls and ellipsoids); New Results and New Trends in Computer Science, ed. H. Maurer; LNCS 555." (1991).
```

Reply:
Thank you for your review!
Actually, this implementation is referenced from Computer Geometry (https://amzn.asia/d/0HX8Tla), and no original implementation on the Internet. There are many someone's reference implementations.
I have not strictly compared which algorithm is the best in our case. Do you think the comparison is needed?